src/dotnet.ts

```diff
@@ -0,0 +1,105 @@
+ *--------------------------------------------------------------------------------------------*/
+'use strict';
+
+import * as vscode from 'vscode';
+import * as which from 'which';
+import * as cp from 'child_process';
+import * as semver from 'semver';
+import * as path from 'path';
+
+export type DotNetInfo = {path: string, version: string};
+export type DotNetCommandResult = {code: number, msg: string};
+
+let dotnetInfo: DotNetInfo | undefined = undefined;
+
+function promptToInstallDotNetCoreSDK(msg: string): void {
+    let installItem = 'Install .NET Core SDK...';
+    vscode.window
+        .showErrorMessage(msg, installItem)
+        .then(
+            (item) => {
+                if (item === installItem) {
+                    vscode.env.openExternal(vscode.Uri.parse('https://dotnet.microsoft.com/download'));
+                }
+            }
+        );
+}
+
+/**
+ * Returns the path to the .NET Core SDK, or prompts the user to install
+ * if the SDK is not found.
+ */
+export function findDotNetSdk(): Promise<DotNetInfo> {
+    return new Promise((resolve, reject) => {
+
+        if (dotnetInfo === undefined) {
+            try {
+                let path = which.sync('dotnet');
+                cp.exec(
+                    `"${path}" --version`,
+                    (error, stdout, stderr) => {
+                        if (error === null) {
+                            dotnetInfo = {path: path, version: stdout.trim()};
+                            resolve(dotnetInfo);
+                        } else {
+                            reject(error);
+                        }
+                    }
+                );
+            } catch (ex) {
+                promptToInstallDotNetCoreSDK('The .NET Core SDK was not found on your PATH.');
+                reject(ex);
+            }
+        } else {
+            resolve(dotnetInfo);
+        }
+
+    });
+}
+
+export function requireDotNetSdk(version?: string): Promise<DotNetInfo> {
+    return new Promise((resolve, reject) => {
+        findDotNetSdk()
+            .then(
+                dotnet => {
+                    if (version !== undefined && semver.lt(dotnet.version, version)) {
+                        let msg = `The PostgreSQL extension requires .NET Core SDK version ${version} or later, but ${dotnet.version} was found.`;
+                        promptToInstallDotNetCoreSDK(msg);
+                        reject(msg);
+                    }
+                    resolve(dotnet);
+                }
+            );
+    });
+}
+
+export function runDotNetCommand(dotNetSdk: DotNetInfo, args: string[]): Promise<DotNetCommandResult> {
+    let dotnetResult: DotNetCommandResult;
+    return new Promise<DotNetCommandResult>((resolve, reject) => {
+        let cmd = 'dotnet';
+        const chunks = [] as any;
+
+        let dotnet = cp.spawn(cmd, args, { cwd: path.dirname(dotNetSdk.path), env: process.env });
```

Comment:
No it does not open a new terminal window. This would just spawn another async child process . It would not run past the lifetime of ADS. I am trying to see how we can do this so that we can handle cancel, stuck builds etc.

Reply:
Actually why does it have to be a separate terminal? The behavior we have right now is similar to what Visual studio does. All the output from the build will be visible in the output terminal and it will run a new process outside which is what vs does as well. Any benefit of opening a new window if we can handle cancel.

Reply:
By "new window" I meant new Terminal window inside vscode (ADS) itself, not another console window. That's how I used to see node builds running:
![image](https://user-images.githubusercontent.com/11728682/67263962-87906080-f45e-11e9-8a4d-1f95b55cdc94.png)

I think if we do this as a background process with no window and redirect stdout/stderr to the Output window of vscode - that would work too. We may need to do extra work to allow cancelling build and making sure they don't run two build in parallel (or build and clean, etc.)

Reply:
This is not integrated with commands but with tasks. When you run a task it can be integrated with the terminal. Somehow I missed your email but I replied to it for task support in ADS. They removed that feature from ADS.

Reply:
Ok, I see.. So currently we print out the output in the ```OUTPUT``` panel then? I think it's fine, we just need to do extra work to support cancellation, etc.
I know in DAMT, we have a progress indicator on the toast in the bottom left when you run the analysis - that allows you to cancel it. I'm thinking, if we should have the same here. Otherwise it's not clear where cancel button will be.